samtranslator/model/sam_resources.py

```diff
@@ -121,6 +121,7 @@ class SamFunction(SamResourceMacro):
         # Intrinsic functions in value of Alias property are not supported, yet
         "AutoPublishAlias": PropertyType(False, one_of(IS_STR)),
         "AutoPublishCodeSha256": PropertyType(False, one_of(IS_STR)),
+        "AutoPublishAliasAdditionalProperties": PropertyType(False, is_type(bool)),
```

Comment:
We shouldn't use `PropertyType`: https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines

Intrinsic resolution is no longer handled by the transform, and `PropertyType` introduces sneaky bugs like in this case. For example, if a customer passes an `!If` that should resolve to false, it'll behave as if it was true anyway (because any non-empty dict is truthy).